src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1246 @@
+        computed
+
+        EXAMPLES::
+
+            sage: # needs sirocco
+            sage: H.<x, y, z> = ProjectivePlaneCurveArrangements(QQ)
+            sage: A = H(y^2 + x*z, y + x - z, x)
+            sage: A.fundamental_group().sorted_presentation()
+            Finitely presented group < x0, x1 | x1^-1*x0^-1*x1*x0 >
+            sage: A.meridians()
+            {0: [x1], 1: [x0], 2: [x1^-1*x0^-1*x1^-1]}
+            sage: A = H(y^2 + x*z, z, x)
+            sage: A.fundamental_group()
+            Finitely presented group < x0, x1 | (x1*x0)^2*(x1^-1*x0^-1)^2 >
+            sage: A.meridians()
+            {0: [x0, x1*x0*x1^-1], 1: [x0^-1*x1^-1*x0^-1], 2: [x1]}
+            sage: A = H(y^2 + x*z, z*x, y)
+            sage: A.fundamental_group()
+            Finitely presented group < x0, x1, x2 | x2*x0*x1*x0^-1*x2^-1*x1^-1,
+                                                    x1*(x2*x0)^2*x2^-1*x1^-1*x0^-1*x2^-1*x0^-1 >
+            sage: A.meridians()
+            {0: [x0, x2*x0*x2^-1], 1: [x2, x0^-1*x2^-1*x1^-1*x0^-1], 2: [x1]}
+
+        This function needs
+        :func:`ProjectivePlaneCurveArrangements.fundamental_group`
+        with the same options, where some examples are shown.
+
+        .. WARNING::
+
+            This functionality requires the sirocco package to be installed.
+        """
+        if not simplified:
+            computed = bool(self._meridians)
+        else:
+            computed = bool(self._meridians_simplified)
+        if not computed:
+            _ = self._fundamental_group(simplified=simplified)
+        if not simplified:
+            return self._meridians
+        return self._meridians_simplified
+
+
+class AffinePlaneCurveArrangements(Parent, UniqueRepresentation):
+    """
+    Affine curve arrangements.
+
+    INPUT:
+
+    - ``base_ring`` -- ring; the base ring
+
+    - ``names`` -- tuple of strings; the variable names
+
+    EXAMPLES::
+
+        sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+        sage: H(x, y^2, x-1, y-1)
+        Arrangement (x, y^2, x - 1, y - 1) in Affine Space
+        of dimension 2 over Rational Field
+    """
+    Element = AffinePlaneCurveArrangementsElement
+
+    def __init__(self, base_ring, names=tuple()):
+        """
+        Initialize ``self``.
+
+        TESTS::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: K = AffinePlaneCurveArrangements(QQ, names=('x', 'y'))
+            sage: H is K
+            True
+            sage: type(K)
+            <class 'sage.schemes.curves.plane_curve_arrangement.AffinePlaneCurveArrangements_with_category'>
+
+        TESTS::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: TestSuite(H).run()
+        """
+        if base_ring not in _Fields:
+            raise ValueError('base ring must be a field')
+        super().__init__(category=Sets())
+        self._base_ring = base_ring
+        self._names = names
```

Comment:
```suggestion
        super().__init__(base_ring, names=names, category=Sets())
```
as `Parent` already provides its own way to store and access these.

Reply:
I tried it but I got plenty of errors.

Reply:
However, this is what you are supposed to do as `Parent` explicitly takes these as arguments and sets these values. You will likely have to replace `self._names` -> `self.variable_names()` and `self._base_ring` -> `self.base_ring()`. If there are other errors after such changes, we should investigate.

Reply:
It seems to work but repeating the same thing for the other class does not work.

Reply:
It should work for all of the classes; it is very general functionality.

Reply:
It works now.